src/frontend/utils/chat/chat.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { converse } from 'utils/window';
+import {
+  generateAnonymousNickname,
+  generateAnonymousToken,
+  getNameFromJID,
+} from './chat';
+
+jest.mock('utils/window', () => ({
+  converse: {
+    env: {
+      Strophe: {
+        getResourceFromJid: jest.fn(),
+      },
+    },
+  },
+}));
+
+describe('utils/chat/chat', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('generates an anonymous nickname.', () => {
+    const generatedNickname = generateAnonymousNickname();
+    expect(generatedNickname.length).toBe(18);
+    expect(generatedNickname.startsWith('Anonymous-')).toBe(true);
+  });
+
+  it('generates an anonymous token.', () => {
+    const generatedToken = generateAnonymousToken();
+    expect(generatedToken.length).toBe(50);
+  });
+
+  it('extracts an username out of a provided JID.', () => {
+    getNameFromJID('id_of_the_room@conference.prosody/JohnDoe');
+    expect(converse.env.Strophe.getResourceFromJid).toHaveBeenNthCalledWith(
+      1,
+      'id_of_the_room@conference.prosody/JohnDoe',
+    );
+  });
```

Comment:
If you use `toHaveBeenNthCalledWith` you should also check how maby times the function was called ?

Reply:
done in [94260d](https://github.com/openfun/marsha/pull/1276/commits/94260daa50c03f34070086bde57fb2de9a43b5f8)